src/main/java/com/synopsys/integration/alert/web/security/AuthorizationManager.java

```diff
@@ -0,0 +1,111 @@
+
+@Component
+public class AuthorizationManager {
+
+    private final AuthorizationUtil authorizationUtil;
+    private final Map<Object, PermissionMatrixModel> permissionCache;
+
+    @Autowired
+    public AuthorizationManager(final AuthorizationUtil authorizationUtil) {
+        this.authorizationUtil = authorizationUtil;
+        permissionCache = new HashMap<>();
+    }
+
+    public boolean hasCreatePermission(final PermissionKeys permissionKey) {
+        return currentUserHasPermission(permissionKey, AccessOperation.CREATE);
+    }
+
+    public boolean hasDeletePermission(final PermissionKeys permissionKey) {
+        return currentUserHasPermission(permissionKey, AccessOperation.DELETE);
+    }
+
+    public boolean hasReadPermission(final PermissionKeys permissionKey) {
+        return currentUserHasPermission(permissionKey, AccessOperation.READ);
+    }
+
+    public boolean hasWritePermission(final PermissionKeys permissionKey) {
+        return currentUserHasPermission(permissionKey, AccessOperation.WRITE);
+    }
+
+    public boolean hasExecutePermission(final PermissionKeys permissionKey) {
+        return currentUserHasPermission(permissionKey, AccessOperation.EXECUTE);
+    }
+
+    public boolean currentUserHasPermission(final PermissionKeys permissionKey, final AccessOperation operation) {
+        final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        final PermissionMatrixModel permissions;
+        if (permissionCache.containsKey(authentication.getPrincipal())) {
+            permissions = permissionCache.get(authentication.getPrincipal());
+        } else {
+            permissions = loadPermissionsIntoCache(authentication);
+        }
+        return permissions.hasPermission(permissionKey.getDatabaseKey(), operation);
+    }
+
+    public PermissionMatrixModel loadPermissionsIntoCache() {
+        return loadPermissionsIntoCache(SecurityContextHolder.getContext().getAuthentication());
+    }
+
+    public PermissionMatrixModel loadPermissionsIntoCache(final Authentication authentication) {
+        final PermissionMatrixModel permissions;
+        if (null == authentication || !authentication.isAuthenticated()) {
+            permissions = new PermissionMatrixModel(Map.of());
+        } else {
+            final List<String> roleNames = authentication.getAuthorities().stream()
+                                               .map(GrantedAuthority::getAuthority)
+                                               .filter(role -> role.startsWith(UserModel.ROLE_PREFIX))
+                                               .map(role -> StringUtils.substringAfter(role, UserModel.ROLE_PREFIX)).collect(Collectors.toList());
+            permissions = authorizationUtil.mergePermissionsForRoles(roleNames);
+            permissionCache.put(authentication.getPrincipal().hashCode(), permissions);
+        }
+        return permissions;
```

Comment:
I generally prefer making your if check at the beginning and immediately returning new PermissionMatrixModel instead of an if else.